### PR TITLE
fix qreduceAssay qranges issue

### PR DESCRIPTION
**Pull request by lgeistlinger**
Fixes qreduceAssay qranges issue as described here:

https://github.com/Bioconductor/RaggedExperiment/issues/20#issue-357469794

and as discussed with @LiNk-NY   

Also updates the definition of `weightedmean` as the `simplifyReduce` function in man page and vignette.

### CODE REVIEW

File: inst/scripts/assay-functions-Ex.R [hunk 1]
```diff
@@ -7,8 +7,13 @@ re4 <- RaggedExperiment(GRangesList(
 query <- GRanges(c("chr1:1-14:-", "chr2:11-18:+"))
 
 weightedmean <- function(scores, ranges, qranges)
+{
     ## weighted average score per query range
-    sum(scores * width(ranges)) / sum(width(qranges))
+    ## the weight corresponds to the size of the overlap of each 
+    ## overlapping subject range with the corresponding query range  
+    isects <- pintersect(ranges, qranges)
+    sum(scores * width(isects)) / sum(width(isects))
```
Comment (LiNk-NY) [inline inst/scripts/assay-functions-Ex.R:15]: @lgeistlinger Does it have to be the `sum` of the widths? Wouldn't that give you the total sum of widths?

Comment (lgeistlinger) [reply inst/scripts/assay-functions-Ex.R:15]: Thanks for checking.

Note that `isects` is a `GRangesList` storing the indivual overlaps of each overlapping subject ranges with the query ranges. Accordingly `sum(width(isects))` now returns an integer vector parallel (ie of the same length with 1-to-1 correspondence between elements) to `scores`/ `ranges` / `qranges`. That is differnt to the previous faulty `weightedmean` where `sum(width(qranges))` returned an integer scalar across all `qranges`.

Further note that for the example

```
re4 <- RaggedExperiment(GRangesList(
    GRanges(c(A = "chr1:1-10:-", B = "chr1:8-14:-", C = "chr2:15-18:+"),
        score = 3:5),
    GRanges(c(D = "chr1:1-10:-", E = "chr2:11-18:+"), score = 1:2)
), colData = DataFrame(id = 1:2))

query <- GRanges(c("chr1:1-14:-", "chr2:11-18:+"))
```

the new def of `weightedmean` now correctly returns
```
> qreduceAssay(re4, query, weightedmean)
                 [,1] [,2]
chr1:1-14:-  3.411765    1
chr2:11-18:+ 5.000000    2
```

- the first sample has two ranges (of score 3 and 4) overlapping with the first query range `chr1:1-14:-`. The overlap / weight of the first range (score 3) is 10 bp, for the second range (score 4) it is 7 bp. Accordingly the weighted mean tends slightly more towards 3.
- for the other cases there is only one overlapping range per sampe and query range, and accordingly returns the score of this range.

Comment (LiNk-NY) [reply inst/scripts/assay-functions-Ex.R:15]: Thanks for the clarification. I see that `isects` returns the intersected ranges so using `sum` on the `width`s makes sense.

**Review by LiNk-NY — APPROVED** [review]
Thanks Ludwig @lgeistlinger for providing a fix to this `weightedmean` example functional.
It was tough getting one's head around it.

**Pull request merged**